middleware/user.go

```diff
@@ -11,21 +11,35 @@ import (
 
 func UserCatcher(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		xRhIdentity, ok := c.Get(h.PARSED_IDENTITY).(*identity.XRHID)
-		if !ok {
-			return fmt.Errorf("failed to fetch the identity header")
-		}
-
 		tenantId, ok := c.Get(h.TENANTID).(int64)
 		if !ok {
 			return fmt.Errorf("failed to pull tenant from request")
 		}
 
-		userID := xRhIdentity.Identity.User.UserID
-		if userID != "" {
-			user, err := dao.GetUserDao(&tenantId).FindOrCreate(userID)
+		userIDFromContext := ""
+
+		switch {
+		case c.Get(h.PSK_USER) != nil:
+			userID, ok := c.Get(h.PSK_USER).(string)
+			if !ok {
+				return fmt.Errorf("failed to pull tenant from request")
```

Comment:
I think in this case the error would be `failed to pull user id from request`

Reply:
good eye! thanks